### PR TITLE
Tc2 login with correct email and password

### PR DESCRIPTION
**Pull request by MariamAbduallah**
- The following PR contains the automation of TC#2 login with correct email and password.
Steps needs to be validated: 
1. Launch browser
2. Navigate to url ['http://automationexercise.com'](https://automationexercise.com/)
3. Verify that home page is visible successfully
4. Click on 'Signup / Login' button
5. Verify 'Login to your account' is visible
6. Enter correct email address and password
7. Click 'login' button
8. Verify that 'Logged in as username' is visible
9. Click 'Delete Account' button
10. Verify that 'ACCOUNT DELETED!' is visible

Link to the workflow run file [https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/7816197748/workflow](url)

### CODE REVIEW

File: src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        new SignupLoginPage(driver)
+                .validateOnLoginVisibility(loginTestData.getTestData("Messages.Login"))
+                .registeredUserLogin(registerTestData.getTestData("UserMail.ApiTimeStamp")+ timeStamp + "@gizasystems.com",registerTestData.getTestData("UserPassword"));
+        new NavigationBar(driver)
+                .validateTheLoggedInUser(registerTestData.getTestData("UserName"));
+        new ApisAccountManagement(api)
+                .deleteUserAccount(registerTestData.getTestData("UserMail.ApiTimeStamp")+ timeStamp + "@gizasystems.com",registerTestData.getTestData("UserPassword"))
+                .validateDeleteUser()
+                .validateUserNotFound(registerTestData.getTestData("UserMail.ApiTimeStamp")+ timeStamp + "@gizasystems.com");
+
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java:40]: Please this empty line

File: src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java:43]: Pleas remove one empty line of those

File: src/main/java/com/gizasystems/automationexercise/pages/SignupLoginPage.java [hunk 3]
```diff
@@ -35,6 +39,14 @@ public SignupLoginPage newUserSignup(String name, String email) {
                 .click(signup_button);
         return this;
     }
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/SignupLoginPage.java:41]: Please add empty line here

File: src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    //////////////////// Configurations \\\\\\\\\\\\\\\\\\\\
+    @BeforeClass
+    public void beforeClass() {
+        registerTestData = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/RegisterUser.json");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/login/LoginWithCorrectEmailAndPasswordTest.java:47]: We should only have 1 to 1 relationship between the test data json file and test class.
Please add the registration data in the corresponding test data json file with unique names to the test case.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
Please check the pipeline execution link as it's not working.

**Comment by MariamAbduallah** [comment]
> Please check the pipeline execution link as it's not working.

it works for me.. what was the problem ? , try to copy and paste it in a new tab an let me know.

**Comment by MahmoudElSharkawyGS** [comment]
@MariamAbduallah 
Please check the conflicts and resolve them.
Please check the pipeline execution as no test cases were executed; And please execute it again.

**Comment by MahmoudElSharkawyGS** [comment]
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/7885380784

**Review by MahmoudElSharkawyGS — APPROVED** [review]

**Pull request merged**